skillmap/src/lib/codeCarryover.ts

```diff
@@ -181,4 +193,94 @@ function mergeJRES(previous: string, next: string) {
     }
 
     return JSON.stringify(nextParsed);
+}
+
+function appendTemporaryAssets(blocks: string, assets: string) {
+    if (!blocks) return assets;
+
+    let jres = pxt.U.jsonTryParse(assets) as pxt.Map<Partial<pxt.JRes> | string> || {};
+    if (!jres["*"]) {
+        (jres as any)["*"] = {
+            "mimeType": "image/x-mkcd-f4",
+            "dataEncoding": "base64",
+            "namespace": pxt.sprite.IMAGES_NAMESPACE
+        };
+    }
+
+    // Regex image literals from blocks, append to existing JRes
+    let index = 0;
+    getImages(blocks)
+        .forEach(data => {
+            const id = guidGen();
+            while (jres[`${pxt.sprite.IMAGES_NAMESPACE}.${pxt.sprite.IMAGE_PREFIX}${index}`]) {
+                index++;
+            }
+            jres[id] = {
+                data: pxt.sprite.base64EncodeBitmap(data),
+                mimeType: pxt.IMAGE_MIME_TYPE,
+                displayName: `${pxt.sprite.IMAGES_NAMESPACE}.${pxt.sprite.IMAGE_PREFIX}${index}`
+            }
+            index++;
+        });
+
+    // Regex animations (array of image literals) from blocks, append to existing JRes
+    index = 0;
+    getAnimations(blocks)
+        .forEach(anim => {
+            const id = guidGen();
+            while (jres[`${pxt.sprite.ANIMATION_NAMESPACE}.${pxt.sprite.ANIMATION_NAMESPACE}${index}`]) {
+                index++;
+            }
+            jres[id] = {
+                data: pxt.sprite.encodeAnimationString(anim.frames, anim.interval),
+                mimeType: pxt.ANIMATION_MIME_TYPE,
+                displayName: `${pxt.sprite.ANIMATION_NAMESPACE}.${pxt.sprite.ANIMATION_PREFIX}${index}`
+            }
+            index++;
+        });
+
+    return JSON.stringify(jres)
+}
+
+/**
+ *  <field name="img">img`
+ *      . .
+ *      . .
+ *  `</field>
+ */
+function getImages(text: string): pxt.sprite.BitmapData[] {
```

Comment:
hmmm.. I wonder if we should hardcode the field name as that is part of the block definition and might not always be "img". For example, if someone made an extension that uses images